### ISSUE TITLE
valid email address deemed invalid

### ISSUE THREAD

**Issue by janwijbrand** [pypa/warehouse#3733]
When trying to add an additional email address to my account, the email address is rejected. This email address however /is/ valid. The domain for the email address is `n--tree.net`. I guess the validator trips over the (admittedly uncommon) double dash in it. Hopefully this can be resolved. 

---

**Good First Issue**: This issue is good for first time contributors. If you've already contributed to Warehouse, work on [another issue without this label](https://github.com/pypa/warehouse/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22good+first+issue%22) instead. If there is not a corresponding pull request for this issue, it is up for grabs. For directions for getting set up, see our [Getting Started Guide](https://warehouse.pypa.io/development/getting-started/). If you are working on this issue and have questions, feel free to ask them here, [`#pypa-dev` on Freenode](https://webchat.freenode.net/?channels=%23pypa-dev), or the [pypa-dev mailing list](https://groups.google.com/forum/#!forum/pypa-dev).

**Comment by di** [comment]
@janwijbrand Sorry you're having troubles! This seems to be due to one of our dependencies, `wtforms`, and there isn't really a great way for us to override the behavior.

Would you raise the issue in their tracker instead? https://github.com/wtforms/wtforms/issues. Thanks for the report!

**Comment by janwijbrand** [comment]
Thanks @di. I've made the report here: https://github.com/wtforms/wtforms/issues/384.

**Comment by davidism** [comment]
It's absolutely possible to change the validator you're using. Although it's sort of futile to try to validate addresses rather than just sending them. I'd recommend removing the validator, or replacing it with a simpler one. I don't really like to keep changing the email regex, since it never seems to end. I also don't really have time to release a new WTForms version any time soon.

**Comment by davidism** [comment]
On a related note, it appears I don't have admin access to the WTForms projects on PyPI anymore. I just went through a year long process to get them from the previous maintainer, and they appear to have disappeared with the transition to Warehouse.

**Comment by janwijbrand** [comment]
Thanks @davidism for your reply. I hope @di can reopen the issue and followup on your suggestions?

**Comment by di** [comment]
@davidism Seems to me like the user https://pypi.org/user/davidism/ is still an Owner of the project https://pypi.org/project/WTForms/, do you not see this project on login?

@janwijbrand Could you email me at `di@python.org` from the address you'd like to add?

Ultimately removing the validator entirely is blocked on https://github.com/pypa/warehouse/issues/3632.

**Comment by di** [comment]
This issue is no longer blocked -- all new emails must be verified before they can be used, so we can remove the `wtforms.validators.Email` validator and replace it with something much simpler, like `wtforms.validators.Regexp(r".+@.+\..+")`, which should be good enough to catch most common mistakes.

**Comment by davidism** [comment]
As a side note, WTForms also moved to using the email_validator library, although I don't think there's going to be a release any time soon. But you might want to look into using it too if you want more thorough validation still.

**Comment by robindboer** [comment]
With https://github.com/pypa/warehouse/pull/6013 merged, I think this issue is resolved and can be closed.

**Comment by brainwane** [comment]
@robindboer Thanks for the fix! Do you have an idea of what you might work on next? If you'd like to keep working on Warehouse and you would like a suggestion, please feel free to comment and ask for one :-)

**Comment by robindboer** [comment]
@brainwane Would love to continue to work on Warehouse. I was looking for a next issue already but if you have any suggestions, they are welcome. Thanks!

**Comment by brainwane** [comment]
@robindboer is #5825 something you would be interested in working on?

**Comment by brainwane** [comment]
Actually @robindboer looks like some of my colleagues want to take care of that in a particular way. Are you potentially interested in learning some ElasticSearch and working on #4006 or one of our other issues tagged "search"?

**Comment by robindboer** [comment]
@brainwane thanks for the suggestions. I will look into the issues later today and see if I can pick up any of the issues. I have never worked with ES before but I see that as an opportunity 😄 Thanks!

**Comment by brainwane** [comment]
Thanks, @robindboer - if you're having any trouble, please do feel free to leave a question on the issue, chat with us in IRC in [`#pypa-dev` on Freenode](https://webchat.freenode.net/?channels=%23pypa-dev), or ask a question on the [pypa-dev mailing list](https://groups.google.com/forum/#!forum/pypa-dev).
